File: 134. Gas Station/134. Gas Station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int n = gas.length;
        int totalSurplus = 0;
        int currentSurplus = 0;
        int startStation = 0;

        for (int i = 0; i < n; i++) {
            totalSurplus += gas[i] - cost[i];
            currentSurplus += gas[i] - cost[i];

            if (currentSurplus < 0) {
                startStation = i + 1;
                currentSurplus = 0;
            }
        }

        return totalSurplus >= 0 ? startStation : -1;
    }
}

File: 134. Gas Station/134. Gas Station.rs
impl Solution {
    pub fn can_complete_circuit(gas: Vec<i32>, cost: Vec<i32>) -> i32 {
        let n = gas.len();
        let mut total_surplus = 0;
        let mut current_surplus = 0;
        let mut start_station = 0;

        for i in 0..n {
            total_surplus += gas[i] - cost[i];
            current_surplus += gas[i] - cost[i];

            if current_surplus < 0 {
                start_station = i + 1;
                current_surplus = 0;
            }
        }

        if total_surplus >= 0 { start_station as i32 } else { -1 }
    }
}